src/sidebar/components/menu-item.js

```diff
@@ -56,49 +55,102 @@ export default function MenuItem({
   const leftIcon = isSubmenuItem ? null : renderedIcon;
   const rightIcon = isSubmenuItem ? renderedIcon : null;
 
-  return (
-    <Fragment>
-      {/* FIXME-A11Y */}
-      {/* eslint-disable-next-line jsx-a11y/role-supports-aria-props */}
+  // menuItem can be either a link or a button
+  let menuItem;
+  const hasSubmenuVisible = typeof isSubmenuVisible === 'boolean';
+  const isRadioButtonType = typeof isSelected === 'boolean';
+
+  const onCloseSubmenu = event => {
+    if (onToggleSubmenu) {
+      onToggleSubmenu(event);
+    }
+    // The focus won't work without delaying rendering.
+    setTimeout(() => {
+      menuItemRef.current.focus();
```

Comment:
I still don't have a fix for focus other than this delay

Reply:
I see what you mean. I can't immediately see why this is happening. As long as the reason for having this timeout is documented in the code, I think we can live with it for now until one of us has time to debug this more thoroughly.